src/EnergyPlus/ZoneEquipmentManager.cc

```diff
@@ -1349,6 +1319,44 @@ void SetUpZoneSizingArrays(EnergyPlusData &state)
         TotPeopleInZone = 0.0;
         Real64 ZoneMinOccupancy = 0.;
         ZoneIndex = state.dataSize->FinalZoneSizing(CtrlZoneNum).ActualZoneNum;
+        int DSOAPtr = state.dataSize->FinalZoneSizing(CtrlZoneNum).ZoneDesignSpecOAIndex; // index to DesignSpecification:OutdoorAir object
+        if (DSOAPtr > 0) {
+            auto &thisOAReq = state.dataSize->OARequirements(DSOAPtr);
+            // If this is a DesignSpecification:OutdoorAir:SpaceList check to make sure spaces are valid and belong to this zone
+            if (thisOAReq.numDSOA > 1) {
```

Comment:
So why not fill dsoaSpaceIndexes regardless of how many DesignSpecification:OutdoorAir objects there are? Spaces and zones are already filled out by now, right? If there is 1 DSOA then there is 1 dsoaSpaceIndexes. I say this because those if (this->numDSOA == 1) in the new functions look unnecessary (e.g., OARequirementsData::calcOAFlowRate).